pyensign/utils/tasks.py

```diff
@@ -0,0 +1,36 @@
+import asyncio
+
+
+class WorkerPool:
```

Comment:
This is impressive code - and I'm glad you have good unit tests for it; I'm dealing with deadlocks in the Go code and I could see this being similarly difficult if not maintained.